pkg/install/error_test.go

```diff
@@ -0,0 +1,219 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_resources "github.com/Azure/ARO-RP/pkg/util/mocks/azureclient/mgmt/resources"
+)
+
+func TestIsResourceQuotaExceededError(t *testing.T) {
+	for _, tt := range []struct {
+		name        string
+		inputErr    error
+		wantBool    bool
+		wantMessage string
+	}{
+		{
+			name:     "Is not a DetailedError",
+			inputErr: errors.New("☹"),
+		},
+		{
+			name:     "Does not contain original error",
+			inputErr: autorest.NewErrorWithError(nil, "", "", nil, ""),
+		},
+		{
+			name:     "Original is not a ServiceError",
+			inputErr: autorest.NewErrorWithError(errors.New("☹"), "", "", nil, ""),
+		},
+		{
+			name: "No code in ServiceError",
+			inputErr: autorest.NewErrorWithError(&azure.ServiceError{
+				Details: []map[string]interface{}{{
+					"code":    nil,
+					"message": nil,
+				}}}, "", "", nil, ""),
+		},
+		{
+			name: "Different code in ServiceError",
+			inputErr: autorest.NewErrorWithError(&azure.ServiceError{
+				Details: []map[string]interface{}{{
+					"code":    "AnotherCode",
+					"message": "Something happened",
+				}}}, "", "", nil, ""),
+		},
+		{
+			name: "Quota exceeded but no message in ServiceError",
+			inputErr: autorest.NewErrorWithError(&azure.ServiceError{
+				Details: []map[string]interface{}{{
+					"code":    "QuotaExceeded",
+					"message": nil,
+				}}}, "", "", nil, ""),
+		},
+		{
+			name: "Quota exceeded",
+			inputErr: autorest.NewErrorWithError(&azure.ServiceError{
+				Details: []map[string]interface{}{{
+					"code":    "QuotaExceeded",
+					"message": "Quota exceeded",
+				}}}, "", "", nil, ""),
+			wantBool:    true,
+			wantMessage: "Quota exceeded",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			isQuota, errMsg := isResourceQuotaExceededError(tt.inputErr)
+			if isQuota != tt.wantBool || errMsg != tt.wantMessage {
+				t.Error(tt.wantBool, tt.wantMessage)
```

Comment:
It makes more sense to print the actual value instead of the expected one, so in the test output you will see the name of the test, line where you called `t.Error` and the actual value. From this you will be easily compare the actual value with the expected value defined in the case.

Reply:
You're right - I'll switch that around. I'll fix any other instances where I've done that as well.

Reply:
> We still need to find an answer [to this question](https://github.com/Azure/ARO-RP/pull/316/#discussion_r397942819),

This bug has been fixed.